internal/parser/scanner/scanner.go

```diff
@@ -215,7 +216,7 @@ func (s *scanner) next() rune {
 }
 
 // peek returns the next rune of the scanners input without consuming it.
-func (s *scanner) peek() rune {
+func (s *scanner) peekRune() rune {
```

Comment:
Would a bounds check make sense? Especially with peek?

Reply:
I think before every `peekRune` is called, the `done()` function is called to ensure the bounds are maintained?

Reply:
Should either be documented or done inside peekRune. Also, the godoc is incorrect, as it states the wrong method name.